koku/providers/gcp/provider.py

```diff
@@ -52,6 +61,15 @@ def update_source_data_source(self, credentials, data_source):
                 if source.billing_source.get("data_source", {}).get("dataset") == data_source.get("dataset"):
                     source_filter = Sources.objects.filter(source_id=source.source_id)
                     source_filter.update(billing_source={"data_source": data_source})
+
+                    provider_uuid = source_filter.first().koku_uuid
```

Comment:
I think I am confused by all of this logic. Why do we filter the sources instead of getting the sources entry? What happens here if the `source_filter` queryset is empty? I presume that `source_filter.first().koku_uuid` will result in an error, won't it?

Reply:
Updated slightly.

Reply:
Since we are using `.filter`, do we even need the try/except?